src/lib/comms/sol-lwm2m-bs-server.c

```diff
@@ -482,10 +492,15 @@ sol_lwm2m_bootstrap_server_delete_object_instance(struct sol_lwm2m_bootstrap_ser
     enum sol_coap_response_code response_code),
     const void *data)
 {
+    enum sol_lwm2m_path_props props;
+
     SOL_NULL_CHECK(server, -EINVAL);
     SOL_NULL_CHECK(client, -EINVAL);
     SOL_NULL_CHECK(path, -EINVAL);
 
+    props = sol_lwm2m_common_get_path_props(path);
+    SOL_EXP_CHECK(props != PATH_HAS_INSTANCE, -EINVAL);
```

Comment:
An empty path `/` or a full object `/999` is accepted for Bootstrap Delete. So, this should be `SOL_EXP_CHECK(props == PATH_HAS_RESOURCE, -EINVAL);`.

For readability only - so it's not weird that we're accepting a property called `...INVALID` -, maybe we could have an additional property `PATH_IS_EMPTY` between `PATH_IS_INVALID` and `PATH_HAS_OBJECT`?